lua/nvim-tree/explorer/filters.lua

```diff
@@ -87,8 +111,14 @@ local function custom(path)
   -- filter custom regexes
   local relpath = utils.path_relative(path, vim.loop.cwd())
   for pat, _ in pairs(M.ignore_list) do
-    if vim.fn.match(relpath, pat) ~= -1 or vim.fn.match(basename, pat) ~= -1 then
-      return true
+    if type(pat) == "function" then
+      if pat(path) then
```

Comment:
Let's give the user all the information here to allow creation of any sort of filter.

We can use `lib` `clone_node` to render the subtree to pass. That contains executable.

Reply:
Is it possible to get the node from the path, at the stage the filter is ran?
I believe the explorer has not yet loaded, so `utils.get_node_from_path` returns `nil`.

I am not very familiar with this codebase, so apologies if I am misunderstanding something.

Reply:
It will often be null; we just have to test for that and ignore it.